File: iosApp/iosApp/Presentation/Navigation/Route.swift
//
//  Route.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/05.
//

import Foundation

enum Route: Hashable {
    case entry
    case main(initialTab: String?)
    case cast(param: String)
    case cafe(param: String)
    case cafeEvent(cafeId: String, eventId: String)
    case cafeDashboard(param: String)
    case banner(cafeId: String?)
    case bannerEdit(cafeId: String?, bannerId: String?)
    case externalLink(title: String, url: String)
    case cafeInfoEdit(param: String?, isRegistrationMode: Bool)
    case noticeEvent(param: String)
    case castEdit(cafeId: String?, castId: String?)
    case schedule(castId: String?)
    case castManagement(cafeId: String, cafeName: String)
    case menuGoods(param: String)
    case menuGoodsEdit(cafeId: String, itemId: String?)
    case reviewEdit(cafeId: String, reviewId: String?)
    case picture(imageUrl: String)
    case checkInMap(initialRegionKey: String?)
    case signIn
    case signUp
    case resetPassword
    case notification
    case settings
    case notificationSettings
    case accountSettings
    case inquiry
    case changePassword
    case community
    case postEdit
    case postDetail(postId: String)
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/navigation/Route.kt
package com.hhp227.concafe.presentation.navigation

import kotlinx.serialization.Serializable

@Serializable
sealed interface Route {
    @Serializable
    data object Entry : Route

    @Serializable
    data class Main(val initialTab: String? = null) : Route

    @Serializable
    data class Cast(val param: String) : Route

    @Serializable
    data class Cafe(val param: String) : Route

    @Serializable
    data class CafeDashboard(val param: String) : Route

    @Serializable
    data class Banner(val cafeId: String? = null) : Route

    @Serializable
    data class BannerEdit(
        val cafeId: String? = null,
        val bannerId: String? = null
    ) : Route

    @Serializable
    data class ExternalLink(val title: String, val url: String) : Route

    @Serializable
    data class CafeInfoEdit(
        val param: String? = null,
        val isRegistrationMode: Boolean = false
    ) : Route

    @Serializable
    data class NoticeEvent(val param: String) : Route

    @Serializable
    data class CastEdit(val cafeId: String? = null, val castId: String? = null) : Route

    @Serializable
    data class Schedule(val castId: String? = null) : Route

    @Serializable
    data class MenuGoods(val param: String) : Route

    @Serializable
    data class MenuGoodsEdit(
        val cafeId: String,
        val itemId: String? = null
    ) : Route

    @Serializable
    data class ReviewEdit(val cafeId: String, val reviewId: String? = null) : Route

    @Serializable
    data class Picture(val imageUrl: String) : Route

    @Serializable
    data object SignIn : Route

    @Serializable
    data object SignUp : Route

    @Serializable
    data object ResetPassword : Route

    @Serializable
    data object Notification : Route

    @Serializable
    data object Settings : Route

    @Serializable
    data object NotificationSettings : Route

    @Serializable
    data object AccountSettings : Route

    @Serializable
    data object Inquiry : Route

    @Serializable
    data object ChangePassword : Route
}
